### ISSUE TITLE
Accessibility Issue With Menu Items

### ISSUE THREAD

**Issue by wplit** [soflyy/oxygen-bugs-and-features#21]
**Describe the bug**
Currently, there is no way to access the sub menu in Oxygen sites for keyboard users. You should be able to tab across the to the sub menu items using a keyboard.

**To Reproduce**
In chrome browser, go to Oxygen's atomic demo page - http://atomic.oxy.host/
Start tabbing across the menu. Notice you can't access the sub menu items with the keyboard alone.

The problem is in the CSS, where you have visibility: hidden; as the way to keep the sub-menu hidden until the user hovers over the menu-item, making the sub menu items not focusable.

**Solution**
Instead of using visibility, you can instead move the sub-menu out of visibility by giving it a right: 5000px; or something similar to push it off screen. Then bring back to right: auto; on both hover and using focus-within for keyboard users. I made a gist with the CSS I've been adding to Oxygen sites to solve the problem. The bottom part is just to match the colors that are set in Oxygen as Oxygen only applies the color changes to hover but not focus.

https://gist.github.com/wplit/9394985fb87ed279ca175cb45a2ee5ef

**Comment by sunnyt7** [comment]
Adding "Accessibility Issue with aos animations in 2.2" to this thread: https://github.com/soflyy/oxygen-bugs-and-features/issues/50

**Comment by adrien-robert** [comment]
+1

**Comment by JayOkey** [comment]
+1

**Comment by thmbaker** [comment]
+1

**Comment by dlovere** [comment]
This is a major need for me, as I am transitioning a suite of sites for a major university from Drupal 7 to Wordpress over the next 16 months. I have been planning on Oxygen for this, but may have to seek a different solution if this is not resolved soon!

**Comment by perkster** [comment]
+1

**Comment by swinggraphics** [comment]
I've been wrestling with every aspect of Oxygen trying to put a site together, and this is absolutely the breaking point for me. I won't be building any sites with Oxygen, and I will be warning folks to stay away until some basics like accessibility are fixed.

**Comment by AlkoKod** [comment]
+1 for Pro Menu element too

**Comment by wpmanagers** [comment]
any more updates on this?

**Comment by AlkoKod** [comment]
> any more updates on this?

I have started using OxyExtras to fix this.

**Comment by wpmanagers** [comment]
> > any more updates on this?
> 
> I have started using OxyExtras to fix this.

can you describe how this was possible? I'm at risk of loosing a client due to lack of accessibility and would love to hear some advice.

**Comment by swinggraphics** [comment]
It is very telling that basic accessibility is labelled "Low Priority" by the Oxygen developer; so low that it hasn't been addressed in any way for two years and counting.

People using assistive technology cannot use a website built with Oxygen without extensive customization from a savvy developer. That is morally wrong, as well as a legal liability.

**Comment by c6410** [comment]
+1

**Comment by gitsimhub** [comment]
+1

**Comment by Mozzarellathicc** [comment]
+1

**Comment by netdoctor** [comment]
+1

**Comment by uhlhosting** [comment]
Folks this requires attention @Spellhammer

**Comment by Spellhammer** [comment]
Added support for tab navigation in 4.1 beta 1. Sorry it took so long.

This still isn't perfect, as the submenus really shouldn't open automatically when you tab to them (e.g. they should be triggered by a button instead), but it's better than it was.

**Comment by wpsumo** [comment]
@Spellhammer when is 4.1 coming, must be soon?

**Comment by Spellhammer** [comment]
@wpsumo 

We have to let the beta breathe for a little while to allow further internal testing & for users to submit bug reports.

Then we'll release the RC which is considered production ready.

Unless some major issue is found that we have to spend a lot of time fixing, 4.1 final should be out within a couple of weeks.

**Comment by wpsumo** [comment]
@Spellhammer Meant the beta, didn't see it but just saw the release notes thanks!

**Comment by Spellhammer** [comment]
@wpsumo Whoops! The beta is available now in the customer portal.

**Comment by AlkoKod** [comment]
@Spellhammer do you add the accessible navigation also to the tabs element? It will be a gamechanger for EDU websites... Thanks!
